libpod/image/image.go

```diff
@@ -1075,3 +1077,176 @@ func (i *Image) Comment(ctx context.Context, manifestType string) (string, error
+	imgs, _ := i.imageruntime.store.Images()
+	layers, _ := i.imageruntime.store.Layers()
+
+	// Builds the layer dependency chain with ParentID and ChildID
+	layerInfoMap := make(map[string]*LayerInfo)
+	for _, layer := range layers {
+		existingLayer, ok := layerInfoMap[layer.ID]
+		if !ok {
+			layerInfoMap[layer.ID] = &LayerInfo{
+				ID:       layer.ID,
+				Size:     layer.UncompressedSize,
+				ParentID: layer.Parent,
+			}
+
+			//Add current Layer as child of Parent layer.
+			existingLayer, ok = layerInfoMap[layer.Parent]
+			if !ok {
+				layerInfoMap[layer.Parent] = &LayerInfo{
+					ID: layer.Parent,
+				}
+				layerInfoMap[layer.Parent].ChildID = append(layerInfoMap[layer.Parent].ChildID, layer.ID)
+			} else {
+				existingLayer.ChildID = append(existingLayer.ChildID, layer.ID)
+			}
+		} else {
+			existingLayer.ParentID = layer.ID
+			existingLayer.Size = layer.UncompressedSize
+		}
+	}
+
+	// Add the Repo Tags to Top layer of each image.
+	for _, img := range imgs {
+		e, ok := layerInfoMap[img.TopLayer]
+		if !ok {
+			return nil, fmt.Errorf("image top-layer not found in LayerList! : %s ", img.TopLayer)
+		}
+		e.RepoTags = append(e.RepoTags, img.Names...)
+	}
+	return layerInfoMap, nil
+}
+
+// Stores all parent layers using which image is built in imageInfo
+func buildImageDependencyMap(i *Image, imageInfo *InfoImage, layerMap map[string]*LayerInfo, layerID string) error {
+	if layerID == "" {
+		return nil
+	}
+	ll, ok := layerMap[layerID]
+	if !ok {
+		return fmt.Errorf("lookup error: layerid  %s, not found", layerID)
+	}
+	if err := buildImageDependencyMap(i, imageInfo, layerMap, ll.ParentID); err != nil {
+		return err
+	}
+
+	imageInfo.Layers = append(imageInfo.Layers, *ll)
+	imageInfo.Size = imageInfo.Size + ll.Size
+	return nil
+}
+
+// Stores all children layers which are created using given Image.
+func buildImageDependentsMap(i *Image, imageInfo *InfoImage, layerMap map[string]*LayerInfo, layerID string, prefixSpace int, buffer *bytes.Buffer) error {
+	if layerID == "" {
+		return nil
+	}
+	ll, ok := layerMap[layerID]
+	if !ok {
+		return fmt.Errorf("lookup error: layerid  %s, not found", layerID)
+	}
+	imageInfo.Layers = append(imageInfo.Layers, *ll)
+	imageInfo.Size = imageInfo.Size + ll.Size
+	for p := 0; p < prefixSpace; p++ {
+		buffer.WriteString(" ")
+	}
+	buffer.WriteString(fmt.Sprintf("└─ ID: %s Size: %v Tags: %s\n", ll.ID[:12], units.HumanSizeWithPrecision(float64(ll.Size), 4), ll.RepoTags))
+	if len(ll.RepoTags) > 0 {
+		prefixSpace++
+	}
+	for _, childID := range ll.ChildID {
+		if err := buildImageDependentsMap(i, imageInfo, layerMap, childID, prefixSpace, buffer); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func formatImageDependencyOutputBuffer(buffer *bytes.Buffer, imageInfo *InfoImage) error {
+	buffer.WriteString(fmt.Sprintf("Image ID     : %s\n", imageInfo.ID[:12]))
+	buffer.WriteString(fmt.Sprintf("RepoTags     : %s\n", imageInfo.RepoTags))
```

Comment:
(It’s not obvious to me that duplicating the ID and tags at the top is necessary, but that’s up to others.)